python/taichi/lang/kernel_impl.py

```diff
@@ -574,6 +577,42 @@ def call_back():
                 callbacks.append(get_call_back(v, gpu_v))
         return tmp, callbacks
 
+    def get_paddle_callbacks(self, v, has_pp):
+        callbacks = []
+
+        def get_call_back(u, v):
+            def call_back():
+                u.copy_(v, False)
+
+            return call_back
+
+        assert has_pp
+        assert isinstance(v, paddle.Tensor)
+
+        tmp = v.value().get_tensor()
+        taichi_arch = self.runtime.prog.config.arch
+
+        if v.place.is_gpu_place():
+            # External tensor on cuda
+            if taichi_arch != _ti_core.Arch.cuda:
+                # copy data back to cpu
+                host_v = v.cpu()
+                tmp = host_v.value().get_tensor()
+                callbacks.append(get_call_back(v, host_v))
```

Comment:
Actually there's one concern about the lifetime of "host_v", which seems to be a local variable here. By the time the functions in "callbacks" are executed, it's important to make sure the "host_v" is still alive. 

Could you do me a favor to print the ref count of host_v inside the call back function (marked above) and check if it's ref_count > 0? Thanks!

Reply:
Thank you for your review. I see what you mean. But I currently can't use the paddlepaddle-gpu on the develop branch on my machine. This prevents me from testing this situation. Sorry for this  :joy:.

Reply:
Hi Zhang,
Thanks for the comments! No worries about the environment, let me try it out then.

> I am more familiar with life cycles in C++ than Python.
Lol, same to me, that's why I'm not so sure whether "host_v"'s reference will be increased in the callback function.